app/services/record_declarations/base.rb

```diff
@@ -86,6 +89,17 @@ def find_or_create_record!
       end
     end
 
+    def record_exist?
```

Comment:
What this really means is `record_exists_and_does_not_match_incoming_request?`, or something to that effect.
I would prefer to relax `find_or_create_record` query, and have a validation done after we get the record.

Reply:
renamed in e3ed1fe